packages/discovery-provider/src/api/v1/helpers.py

```diff
@@ -667,23 +666,6 @@ def __schema__(self):
         return param
 
 
-# Helper to allow consumer to pass message and signature headers as request params
-def add_auth_headers_to_parser(parser, required=True):
-    parser.add_argument(
-        MESSAGE_HEADER,
-        required=required,
```

Comment:
Feels like we should keep this but make them non-required, so that the REST API docs/swagger pick them up?

Reply:
Revitalizing this PR. Yes this is is a good point, but I find it somewhat error prone to remember to call this helper in places, so instead I added this into `@auth_middleware`. 

@schottra also adding you as a reviewer since this is relevant to your current work. Lmk if either of you have concerns.

The gist of this change is removing `add_auth_headers_to_parser` and having it picked up with the middleware

```
class SalesDownload(Resource):
    @ns.expect(sales_download_parser)
    @auth_middleware(sales_download_parser)
    def get(self, id, authed_user_id=None):
         pass
```

I dream for a day when we have one parser/response marshaller/endpoint per file...

Reply:
Love this! It's unfortunate that we have to trade-off optional fields in the SDK with missing documentation on the REST/Swagger side, but I like where it's landed!

Reply:
yeah ugh!